### ISSUE TITLE
Visual overlapping of annotation icon and vertical axis tick/label

### ISSUE THREAD

**Issue by dej611** [elastic/elastic-charts#1400]
**Describe the bug**

We use the annotation system to draw thresholds in Lens, and upon adding the Icon marker associated with the annotation, it was noticed that it is possible a collision of the marker with the vertical axis.

![image](https://user-images.githubusercontent.com/3884767/133116148-cef8c1fe-ae12-4469-a07c-2f0769da6ebc.png)

**Kibana Cross Issues**
https://github.com/elastic/kibana/issues/112793

**Checklist**

Delete any items that are not applicable to this issue.

- [ ] every related Kibana issue is listed under `Kibana Cross Issues` list
- [ ] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present

**Comment by nickofthyme** [comment]
As of right now I believe we leave it to the user to set the value of the annotation _gutter_, if you will, such that it has the width or height it needs. This would be through `theme. chartPaddings`. Is this true @markov00?

In you case you are using the eui theme which removes the axes ticks making this collision more apparent.

I think we could improve this to be automatic in the future but there are cases like annotations outside and adaptive `markBody` that make it more difficult.

![Screen Recording 2021-09-22 at 08 58 51 AM](https://user-images.githubusercontent.com/19007109/134358321-660164b4-3c79-4a5e-8545-a3ea3199c15d.gif)

**Comment by dej611** [comment]
I made some experiments with annotation on the `left`/`right`/`bottom` axes tweaking some parameters.

Initial state with overlap:

<img width="703" alt="Screenshot 2021-09-22 at 19 25 53" src="https://user-images.githubusercontent.com/924948/134392601-62065619-7cd0-4e63-b961-f4cc747b3c4a.png">

Here's using the `chartPaddings` parameter (better, but the axes should shrink together with the chart IMHO):

<img width="684" alt="Screenshot 2021-09-22 at 19 27 18" src="https://user-images.githubusercontent.com/924948/134392789-b8b579de-165c-4305-ba2f-c2403ef39726.png">

Here's using the `tickLabels.padding` (better but the padding is applied in any direction around the label, not just between the label and the axis):

<img width="690" alt="Screenshot 2021-09-22 at 19 26 32" src="https://user-images.githubusercontent.com/924948/134392668-5253d2ce-5e88-4a97-bd32-a79876b1294a.png">

**Comment by markov00** [comment]
Yes @nickofthyme, at the moment this is true, we don't automatically allocate space for the icon on the axis.
Together with the nick workaround, I can suggest to move the icon to the opposite side of the axis to avoid collisions and to have a clean axis.

If you are in the bad situation of having a dual-axis chart (highly discouraged) is advised to keep the icon near the relative axis draw it inside the chart (better if placed on the top/bottom of the line annotation.

Remember that you can also use directly the value instead if just an icon as in the below example.

Finally you can also have the icon externally placed without adding the `tickLabel.padding` but using the `tickLine:{size: sizeoftheicon, visible: true, stroke: 'transparent'}`
this will add a 'transparent' tick line that can be used in replacement of the tick padding.

Also another possibility, that doesn't fully resolve but can help reading the chart is to color the axis with the same color of the annotation line to improve, for non-color blinded person, the link between the axis and the annotation in a dual axis situation




<img width="632" alt="Screenshot 2021-09-22 at 19 41 18" src="https://user-images.githubusercontent.com/1421091/134394935-0d0133f9-3386-4eac-8efa-175caa1d8574.png">

**Comment by markov00** [comment]
you can also use `tickLabel: {padding: { inner: size_of_the_marker } }` instead of `tickLabel:{padding: size_of_the_marker}}` to

**Comment by dej611** [comment]
> you can also use `tickLabel: {padding: { inner: size_of_the_marker } }` instead of `tickLabel:{padding: size_of_the_marker}}` to

I think this one is the best tradeoff at the moment.
<img width="737" alt="Screenshot 2021-09-23 at 11 33 45" src="https://user-images.githubusercontent.com/924948/134485183-6cfecccd-ebc8-4f12-8ab0-183d88150df6.png">

> If you are in the bad situation of having a dual-axis chart (highly discouraged) is advised to keep the icon near the relative axis draw it inside the chart (better if placed on the top/bottom of the line annotation.
This is definitely an interesting idea and something we might consider given https://github.com/elastic/kibana/issues/112921

> Also another possibility, that doesn't fully resolve but can help reading the chart is to color the axis with the same color of the annotation line to improve, for non-color blinded person, the link between the axis and the annotation in a dual axis situation
> 
> <img alt="Screenshot 2021-09-22 at 19 41 18" width="632" src="https://user-images.githubusercontent.com/1421091/134394935-0d0133f9-3386-4eac-8efa-175caa1d8574.png">

This makes sense, but I doubt is compatible with a generic case where the user can have multiple annotations on the same axis with distinct colors as in our use case.

**Comment by dej611** [comment]
Managed to find a solution for all the various combinations (title + tick labels, title only, etc...) in this PR combining all the padding/margin parameters in the Axis component: https://github.com/elastic/kibana/pull/112952
